File: Frontend/src/interfaces/response-dtos/folder-dto.ts
import { AudioDto } from "../creation-dtos/audio-dto"
import { EbookDto } from "../creation-dtos/ebook-dto"
import { ImageDto } from "../creation-dtos/image-dto"
import { VideoDto } from "../creation-dtos/video-dto"


export interface FolderDto {
    id: number
    name: string
    audios?: AudioDto[]
    ebooks?: EbookDto[]
    images?: ImageDto[]
    videos?: VideoDto[]
}

File: Backend/Media_library/Dtos/ResponseDtos/FolderDto.cs
namespace Media_library.Dtos.ResponseDtos;

public class FolderDto
{
    public int Id { get; set; }
    public string Name { get; set; }
    public IEnumerable<AudioDto> Audios { get; set; }
    public IEnumerable<VideoDto> Videos { get; set; }
    public IEnumerable<ImageDto> Images { get; set; }
    public IEnumerable<EbookDto> Ebooks { get; set; }
}